### PR TITLE
RATIS-1412. Add ratis-shell snapshot command

### PR DESCRIPTION
**Pull request by codings-dan**
## What changes were proposed in this pull request?

Add a sub command to take snapshot of the ratis server

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/RATIS-1437

## How was this patch tested?
UT

### CODE REVIEW

**Comment by codings-dan** [comment]
@szetszwo I add the take snapshot command, PTAL, thx!  Now it seems that we can only let the leader take snapshot, should we also support follower to take snapshot?

File: ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/snapshot/TakeSnapshotCommand.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  @Override
+  public String getCommandName() {
+    return "create";
+  }
+
+  @Override
+  public int run(CommandLine cl) throws IOException {
+    super.run(cl);
+    try(final RaftClient raftClient = RaftUtils.createClient(getRaftGroup())) {
+      RaftClientReply reply = raftClient.getSnapshotManagementApi().create(3000);
```
Comment (szetszwo) [inline ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/snapshot/TakeSnapshotCommand.java:52]: The timeout should be a parameter.

File: ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/snapshot/TakeSnapshotCommand.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      processReply(reply, () -> String.format("Failed to take snapshot of peerId %s", raftClient.getLeaderId()));
```
Comment (szetszwo) [inline ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/snapshot/TakeSnapshotCommand.java:53]: It should print out the returned snapshot index.

**Review by szetszwo — COMMENTED** [review]
@codings-dan , some comments inlined.  Please also add some tests.  Thanks.

**Comment by codings-dan** [comment]
@szetszwo Thanks for reviewing code. I have update the code, PTAL again, thx! I will add some unit tests for the whole ratis-shell module in the future patch.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**